client/src/components/ShareButton.js

```diff
@@ -47,6 +51,12 @@ export class ShareButton extends React.Component {
       icon_size,
     } = this.props;
 
+    const common_icon_props = {
+      width: "50px",
+      height: "50px",
+    };
+    const hashtag_replaced_url = _.replace(url, "#", "%23");
```

Comment:
I think you want
```
const fully_encoded_url = encodeURIComponent(url);
```

(note `encodeURI` URL encodes a string with the exception of semanticly meaningful URL characters like ? or #. `encodeURIComponent` encodes everything)